### PR TITLE
Add some additional security related proxy and ssl header configs

### PR DESCRIPTION
**Pull request by nemchik**
See https://observatory.mozilla.org/
After running the check on observatory it offers numerous recommendations for improvements and specific instructions on how to apply each recommendation. These changes come from their recommendations.

[linuxserverurl]: https://linuxserver.io
[linuxserver.io][linuxserverurl]

### CODE REVIEW

File: root/defaults/nginx.conf [hunk 1]
```diff
@@ -74,7 +74,7 @@ http {
 	##
 	include /etc/nginx/conf.d/*.conf;
 	include /config/nginx/site-confs/*;
-  
+
```
Comment (nemchik) [inline root/defaults/nginx.conf]: This was just whitespace removal that my text editor did.

**Comment by nemchik** [comment]
FYI I rebased this PR to remove a duplicate line I had created accidentally.

File: root/defaults/proxy.conf [hunk 2]
```diff
@@ -14,10 +14,13 @@ proxy_connect_timeout 240;
 proxy_set_header Host $host:$server_port;
 proxy_set_header X-Real-IP $remote_addr;
 proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
-proxy_set_header X-Forwarded-Proto https;
+proxy_set_header X-Forwarded-Proto $scheme;
```
Comment (aptalca) [inline root/defaults/proxy.conf]: This one is not necessary. We actually want to send https when we use ssl anyway

File: root/defaults/proxy.conf [hunk 3]
```diff
@@ -14,10 +14,13 @@ proxy_connect_timeout 240;
...
+proxy_set_header X-Forwarded-Host $host;
+proxy_set_header X-Forwarded-Ssl on;
 proxy_redirect  http://  $scheme://;
 proxy_http_version 1.1;
 proxy_set_header Connection "";
+proxy_cookie_path / "/; HTTPOnly; Secure";
```
Comment (aptalca) [inline root/defaults/proxy.conf]: This seems to be a pretty crude approach. Likely to break certain apps who rely on cookies not meant to be httponly. This should be added to specific reverse proxy configs on a case by case basis

Comment (nemchik) [reply root/defaults/proxy.conf]: I believe just having `secure` in there satisfies the scanner (I'll have to test). Would it be acceptable to keep this line but omit `httponly`?

File: root/defaults/proxy.conf (hunk 2 above)
Comment (nemchik) [reply root/defaults/proxy.conf]: That makes sense, this was something I did in other nginx environments to accommodate http when no SSL was being used, but I guess that doesn't really apply here in context.

**Comment by aptalca** [comment]
Jenkins test this please

**Pull request merged**